portfolio/src/main/webapp/js/custom-script.js

```diff
@@ -251,27 +253,69 @@ function allowUserSubmit() {
         }
       });
 }
-google.charts.load('current', {'packages': ['corechart']});
-google.charts.setOnLoadCallback(drawStateIncarcerationChart);
-
-/** Creates chart and adds it to the page. */
-function drawStateIncarcerationChart() {
-  const stateData = new google.visualization.DataTable();
-  stateData.addColumn('string', 'Race');
-  stateData.addColumn('number', 'Count');
-  stateData.addRows([
-    ['Whites', 201],
-    ['Blacks', 1767],
-    ['Hispanics', 385],
-  ]);
-
-  const stateChartDimensions = {
-    'title': 'Incarceration Rates (per 100,000) by Race in California',
-    'width': 500,
-    'height': 400,
+
+drawTimeSeriesChart();
+
+/** Adds a line chart to page showing the global avg temp from a csv */
+async function drawTimeSeriesChart() {
+  const tempData = await getTempData();
+  const chart = {
+    type: 'line',
+    data: {
+      labels: tempData.xAxis,
+      datasets: [{
+        label: 'Mean Temp',
+        backgroundColor: 'rgb(255, 0, 0)',
+        fill: false,
+        borderColor: 'rgb(255, 0, 0)',
+        data: tempData.yAxis,
+      }],
+    },
+
+    options: {
+      reponsive: true,
+      title: {
+        display: true,
+        text: 'Zonal Annual Temperature Mean',
+      },
+      scales: {
+        xAxes: [{
+          display: true,
+          scaleLabel: {
+            display: true,
+            labelString: 'Year',
+          },
+        }],
+        yAxes: [{
+          display: true,
+          scaleLabel: {
+            display: true,
+            labelString: 'Temperature Mean',
+          },
+        }],
+      },
+    },
   };
 
-  const chart = new google.visualization.PieChart(
-      document.getElementById('chart-container'));
-  chart.draw(stateData, stateChartDimensions);
+  const timeSeries = document.getElementById('timeseries').getContext('2d');
+  new Chart(timeSeries, chart);
+
+  async function getTempData() {
+    // Source of Data: https://data.giss.nasa.gov/gistemp/
+    const xAxis = [];
+    const yAxis = [];
+
+    const response = await fetch('globaltemp.csv');
+    const zonalTempData = await response.text();
+
+    const table = zonalTempData.split('\n').slice(1);
+    table.forEach((row) => {
+      const columns = row.split(',');
+      const year = columns[0];
+      xAxis.push(year);
+      const temp = columns[1];
+      yAxis.push(parseFloat(temp) + 14);
```

Comment:
Why do you add 14 to the temperatures.

Reply:
+1

Reply:
Without adding 14, the chart would show the difference between the mean of the zonal annual temp instead of showing just the avg mean. That would've caused negative values, but the source of the data listed that the global mean is 14°C